lib/modules/datasource/gitlab-tags/index.ts

```diff
@@ -17,7 +24,7 @@ export class GitlabTagsDatasource extends Datasource {
     this.http = new GitlabHttp(GitlabTagsDatasource.id);
   }
 
-  override readonly defaultRegistryUrls = [defaultRegistryUrl];
+  override readonly defaultRegistryUrls = getDefaultRegistryUrl;
```

Comment:
```suggestion
  override defaultRegistryUrls(): string[] {
    return ...
  }
```
can't we use a simple override function?

Reply:
not sure either @rarkins WDYT?

Reply:
sure

Reply:
can't be a member

```
lib/modules/datasource/gitlab-tags/index.ts:20:3 - error TS2425: Class 'Datasource' defines instance member property 'defaultRegistryUrls', but extended class 'GitlabTagsDatasource' defines it as instance member function.

20   defaultRegistryUrls(): string[] {
     ~~~~~~~~~~~~~~~~~~~
```

https://github.com/Microsoft/TypeScript/issues/27965

😞 

@JamieMagee Do you have an idea for nice code here?

Reply:
```suggestion
  override get defaultRegistryUrls(): string[] {
    const { platform, endpoint } = GlobalConfig.get();
    return platform === 'gitlab' && endpoint
      ? [endpoint, defaultRegistryUrl]
      : [defaultRegistryUrl];
  }
```
I think we should use a getter function here.

@rarkins Shouldn't we also always add `defaultRegistryUrl` too? It will always use local platform, so we can't use default gitlab.com without a config override.

Reply:
@rarkins this way we don't need to change the interface at all

Reply:
Error
```
'defaultRegistryUrls' is defined as a property in class 'Datasource', but is overridden here in 'GitlabTagsDatasource' as an accessor.
````

Reply:
Only way to use a getter function of same name in child class is by having the same getter in parent class as well.

Reply:
> @rarkins Shouldn't we also always add `defaultRegistryUrl` too? It will always use local platform, so we can't use default gitlab.com without a config override.

Sorry, I don't understand the question

Reply:
I agree the getter change is going to be the cleaner one. It's just going to require you to touch every datasource.

Reply:
let me test something today, will suggest new solution then.

Reply:
OK, there isn't an elegant ways in typescript to override a property with an accessor or vice versa.

So leave this line as is and but move the `getDefaultRegistryUrl` to `./utils.ts`